cartography/intel/github/repos.py

```diff
@@ -12,11 +13,25 @@
 from packaging.utils import canonicalize_name
 
 from cartography.intel.github.util import fetch_all
+from cartography.intel.github.util import PaginatedGraphqlData
 from cartography.util import run_cleanup_job
 from cartography.util import timeit
 
 logger = logging.getLogger(__name__)
 
+
+@dataclass(frozen=False)
+class UserAffiliationAndRepoPermission:
+    """
+    Representation of a user's permission level to a GitHub repo and affiliation type to the target GitHub org. See:
+    - Permission: https://docs.github.com/en/graphql/reference/enums#repositorypermission
+    - Affiliation: https://docs.github.com/en/graphql/reference/enums#collaboratoraffiliation
+    """
+    user: Dict
+    permission: str  # WRITE, MAINTAIN, ADMIN, etc
+    affiliation: str  # OUTSIDE, DIRECT
```

Comment:
I was led to creating this from a [similar thing](https://github.com/cartography-cncf/cartography/blob/master/cartography/intel/github/teams.py#L20) done in teams.py.  There, a namedtuple is used.  Here, I used a dataclass.  Is that confusing, or is it okay?